system/service.py

```diff
@@ -520,13 +520,15 @@ def sysv_is_enabled(name):
             return False
 
     def get_systemd_status_dict(self):
-
         # Check status first as show will not fail if service does not exist
         (rc, out, err) = self.execute_command("%s show '%s'" % (self.enable_cmd, self.__systemd_unit,))
-        if rc != 0:
-            self.module.fail_json(msg='failure %d running systemctl show for %r: %s' % (rc, self.__systemd_unit, err))
-        elif 'LoadState=not-found' in out:
-            self.module.fail_json(msg='systemd could not find the requested service "%r": %s' % (self.__systemd_unit, err))
+        if self.module.params['must_exist']:
+            if rc != 0:
+                self.module.fail_json(msg='failure %d running systemctl show for %r: %s' % (rc, self.__systemd_unit, err))
+            elif 'LoadState=not-found' in out:
+                self.module.fail_json(msg='systemd could not find the requested service "%r": %s' % (self.__systemd_unit, err))
+        else:
+            self.module.exit_json(changed=False, exists=False)
```

Comment:
Not found doesn't exit 0.  Systemctl returns a non 0 exit code when the unit file does not exist.

```
[root@aio-01-rhel-7 ~]# systemctl restart foo
Failed to issue method call: Unit foo.service failed to load: No such file or directory.
[root@aio-01-rhel-7 ~]# echo $?
6
```

Since it is not 0 we get a failure, and bombs out the module.